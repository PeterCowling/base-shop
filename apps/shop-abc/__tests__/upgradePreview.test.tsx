import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import UpgradePreviewPage from "../src/app/upgrade-preview/page";

describe("upgrade preview page", () => {
  it("renders changes, preview links and publish CTA", async () => {
    const fetchMock = jest
      .spyOn(global, "fetch" as any)
      .mockResolvedValueOnce({
        json: async () => ({
          components: [
            {
              file: "molecules/Breadcrumbs.tsx",
              componentName: "Breadcrumbs",
            },
          ],
          pages: ["1"],
        }),
      } as any)
      .mockResolvedValueOnce({
        json: async () => ({ token: "tok" }),
      } as any);

    const { baseElement } = render(<UpgradePreviewPage />);
    await screen.findByText("Breadcrumbs");
    await screen.findByText("Preview 1");
    expect(screen.getByText("Approve & publish")).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
    fetchMock.mockRestore();
  });
});
