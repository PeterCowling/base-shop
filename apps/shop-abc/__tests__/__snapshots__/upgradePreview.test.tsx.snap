// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`upgrade preview page renders changes, preview links and publish CTA 1`] = `
<body>
  <div>
    <div
      class="space-y-8"
    >
      <ul
        class="list-disc pl-4"
      >
        <li>
          Breadcrumbs
        </li>
      </ul>
      <ul
        class="list-disc pl-4"
      >
        <li>
          <a
            href="/preview/1?upgrade=tok"
          >
            Preview 1
          </a>
        </li>
      </ul>
      <button
        class="rounded border px-4 py-2"
        type="button"
      >
        Approve & publish
      </button>
    </div>
  </div>
</body>
`;
